caser/caser.go

```diff
@@ -8,8 +8,68 @@ import (
 	"golang.org/x/text/language"
 )
 
+type Caser struct {
+	initialisms            map[string]bool
+	camelToSnakeExceptions map[string]string
+	snakeToCamelException  map[string]string
+}
+
+type Option func(*Caser)
+
+// WithCustomInitialims allows to specify custom initialisms for caser.
+func WithCustomInitialims(fields map[string]bool) Option {
+	return func(c *Caser) {
+		for k, v := range fields {
+			c.initialisms[k] = v
+		}
+	}
+}
+
+// WithCustomExceptions allows to specify custom exceptions for caser.
+// The parameter is a map of camel:snake values like map[string]string{"oauth":"OAuth"}
+func WithCustomExceptions(fields map[string]string) Option {
+	return func(c *Caser) {
+		for k, v := range fields {
+			c.camelToSnakeExceptions[v] = k
+			c.snakeToCamelException[k] = v
+		}
+	}
+}
+
+// New creates a new instance of caser
+func New(opts ...Option) *Caser {
+	c := &Caser{
+		initialisms:            make(map[string]bool),
+		camelToSnakeExceptions: make(map[string]string),
+		snakeToCamelException:  make(map[string]string),
+	}
+	for k, v := range commonInitialisms {
```

Comment:
I thought a bit about this API, and down the line we may want to provide an option to skip initialisms so that you can generate AWS-style `Ec2InstanceId` instead of Go-Style `EC2InstanceID`, if you need to.

But we don't have this use-case today, and I think we can later provide it with in a backwards-compatible way using a new option, `WithInitialismsDisabled()` or something along those lines. So I think this is fine, but just sharing my thoughts.

Reply:
you can disable an initialism by setting `map[string]bool{"ID":false}`

Reply:
True, but you cannot reliably disable _all_ initialisms unless we expose our internal list (which I'm not saying we should necessarily do). You could copy-paste the list into your own code and set them all to false, but if we add a new initialism it will slip through again.